### ISSUE TITLE
Preprocessors now make schemas order-dependent, breaking form integration.

### ISSUE THREAD

**Issue by Austin-Love** [colinhacks/zod#2900]
TLDR; Preprocessors in React Hook Forms resolvers no longer invalidate until all previous fields without preprocessors are valid.

We just updated from 3.21 to 3.22, and have found a breaking issue with the integration of preprocessors and react hook forms that did not exist in 3.21. 

We have a preprocesser called ZodNum that we use because the default z.coerce.number() translates empty strings into 0. With the update to 3.22, forms that use preprocessors inside their validation schemas, validate from top to bottom in the order of their field declaration in the schema. If you have a field with z.string().min(5) higher in the schema than any z.preprocess(), the preprocess does not invalidate until the string field is completely valid, leading to an issue where the form is invalid, but the resolver reports it as valid. 

In the code sandbox, the first two fields validate correctly, no matter the state of the first field. But if the second field is not valid, or has not been validated, the third and fourth field cannot validate at all. If the second field is valid, all fields can validate. If you change the order of the fields in the parser so that 'someId' is at the bottom, all fields validate as expected. 

Code Sandbox: https://codesandbox.io/p/sandbox/gracious-violet-wytl24?welcome=true

**Comment by viveleroi** [comment]
I just stumbled onto this issue too and am sad to see no response here. Ensuring the order of objects properties is going to be prone to human error and hard to always catch in our code, so I don't know what to do here.

**Comment by FabianWarnecke** [comment]
I think think is solved with version 3.23.0 via #2912:
https://github.com/colinhacks/zod/releases/tag/v3.23.0

@Austin-Love, can you confirm?
